run-tests.py

```diff
@@ -8,9 +8,19 @@
 
 script_dir = os.path.dirname(os.path.realpath(__file__))
 
-def make_test_variant(rule, variant=None, should_match=True):
+def make_test_variant(rule, variant=None, should_match=True, prebuild=False):
     def case():
         attr_path=f'{rule}.{variant}' if variant is not None else rule
+
+        if prebuild:
```

Comment:
I think it would make more sense if nixpkgs-hammer or rust-checks ran this, otherwise it might lead to failure mode/log checks not being run?, e.g. in case when user modifies the derivation and then runs the tool to see if it fixed the complaint.

Reply:
For @r-rmcgibbo / nixpkgs-review purposes, I'm a pretty strong -1 on this. Having nixpkgs-hammer take over responsibility for running `nix build` adds a lot of complexity.
* No negative caching, so `nix build` followed by `nixpkgs-hammer` will rebuild failed drvs
* `nix build` takes a bunch of arguments related to things like timeouts that we would have to add to the nixpkgs-hammer interace
* I run nix-build in a very controlled environment, including with BPF programs intercepting OOMs syscall failures to monitor reasons that might related to build failure (https://github.com/rmcgibbo/eom-enospc-notify).
* Having `nixpkgs-hammer` call `nix-build` will transform it from being a quick "linter" to something that's a lot more complex for me to run.

I'm not opposed to printing out a notice or something if no build log is available.

Reply:
Maybe we should add a warning that it takes the last build log which may be out of date.

Reply:
I see. But I do not think we should limit ourselves just to fast checks, as long as user can see what is happening (build progress bar) and choose which checks they want to run. For example, #65 could turn out quite slow, yet it is a great addition.

We could have `--no-build` flag in which case the check could complain about logs not being available.

Reply:
How would you like to proceed here?

1. We could merge this PR more-or-less as is.
2. We could could make the log-checks print a warning if there's no available log, but otherwise merge more-or-less as is.
3. We could gate the log-checks behind a flag (`--enable stale-substitute`, `--enable no-python-tests`) but have them off by default. If you elect to enable them, then presumably you understand that they require build logs to function (and then it'd be pretty reasonable for them to warn if the build log is missing).
4. We could wait and only merge this feature once someone has written a bunch of new code to have nixpkgs-hammer call `nix build`, but we'd need to add flags to control how/if `nix build` is called, and proper piping of the logs and progress bar and stuff to the screen.
5. Maybe something else.

I'm not particularly interested in option 4 since my use case for nixpkgs-hammer is mostly in @r-rmcgibbo, and if we merge a feature where nixpkgs-hammer calls `nix build`, I'm going to run 100% of my hammer runs with that feature disabled. So my preference would be one of 1, 2, or 3.

Reply:
Number two sounds like best short term option to me.

Reply:
I did number 2.